pkg/client/s3client.go

```diff
@@ -110,6 +111,49 @@ func (c *S3Client) CheckBucketExists(bucketName string) (bool, error) {
 	return false, nil
 }
 
+// To select objects matching regex from src bucket
+func (c *S3Client) SelectObjects(bucketName string, regex string) ([]string, error) {
+	var matched bool
+	var matchedObjects []string
+	err := c.S3Session.ListObjectsPages(&s3.ListObjectsInput{
+		Bucket: &bucketName,
+	}, func(p *s3.ListObjectsOutput, last bool) (shouldContinue bool) {
+		for _, obj := range p.Contents {
+			matched, _ = regexp.MatchString(regex, *obj.Key)
+			if matched {
+				klog.Infof("Object Selected: %s", *obj.Key)
+				matchedObjects = append(matchedObjects, *obj.Key)
+			}
+		}
+		return true
+	})
+	if err != nil {
+		klog.Infof("failed to list objects", err)
+		return nil, err
+	}
+	return matchedObjects, err
+}
+
+//Func CheckBucketLocationConstraint will verify the existence of the bucket in the particular locationConstraint
+func (c *S3Client) CheckBucketLocationConstraint(bucketName string, bucketLocationConstraint string) (bool, error) {
+
+	GetParams := &s3.GetBucketLocationInput{
```

Comment:
No need to have a exportable variable name if not required, Variable name should be getParams rather than `GetParams`

Reply:
Changed